patches/minecraft/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java.patch

```diff
@@ -0,0 +1,11 @@
+--- ../src-base/minecraft/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
++++ ../src-work/minecraft/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
+@@ -35,6 +35,8 @@
+ 
+     public void func_177484_a(int p_177484_1_, int p_177484_2_, int p_177484_3_, IBlockState p_177484_4_)
+     {
++        if (p_177484_4_ instanceof net.minecraftforge.common.property.IExtendedBlockState)
++            p_177484_4_ = ((net.minecraftforge.common.property.IExtendedBlockState) p_177484_4_).getClean();
```

Comment:
Actually this is probably the only place it's needed as this is the actual class that stores the values and what gets asked for the ids to save to the world